pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -630,6 +625,62 @@ class Build {
         }
     }
 
+    def signInstaller(VersionInfo versionData) {
+        if (versionData == null || versionData.major == null) {
+            context.println "Failed to parse version number, possibly a nightly? Skipping installer steps"
+            return
+        }
+
+        context.node('master') {
+            context.stage("sign installer") {
+                if (buildConfig.TARGET_OS == "mac" || buildConfig.TARGET_OS == "windows") {
+                    try {
+                        signInstallerJob(versionData);
+                        context.sh 'for file in $(ls workspace/target/*.tar.gz workspace/target/*.pkg workspace/target/*.msi); do sha256sum "$file" > $file.sha256.txt ; done'
+                        writeMetadata(versionData, false)
+                        context.archiveArtifacts artifacts: "workspace/target/*"
+                    } catch (e) {
+                        context.println("Failed to build ${buildConfig.TARGET_OS} installer ${e}")
+                        currentBuild.result = 'FAILURE'
+                    }
+                }
+            }
+        }
+    }
+
+    private void signInstallerJob(VersionInfo versionData) {
+        def filter = ""
+
+        switch (buildConfig.TARGET_OS) {
+            case "mac": filter = "**/OpenJDK*_mac_*.pkg"; break
+            case "windows": filter = "**/OpenJDK*_windows_*.msi"; break
+            default: break
+        }
+
+        def nodeFilter = "eclipse-codesign"
```

Comment:
worth parameterizing?

Reply:
this is very specific code to us, it's unlikely that anyone else would consume this part of the pipeline